src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -523,6 +552,31 @@ internal static List<TopicPartitionOffsetError> GetTopicPartitionOffsetErrorList
                 .ToList();
         }
 
+        /// <summary>
+        /// Creates and returns a C rd_kafka_topic_partition_list_t * populated by offsets.
+        /// </summary>
+        /// <returns>
+        /// If offsets is null a null IntPtr will be returned, else a IntPtr
+        /// which must destroyed with LibRdKafka.topic_partition_list_destroy()
+        /// </returns>
+        internal static IntPtr GetCTopicPartitionList(ICollection<TopicPartitionOffset> offsets)
```

Comment:
there is code similar to this in quite a number of places, probably be good to remove the duplication at some point (not necessarily in this PR).

I'm not too fond of the C in the name here, though it's internal so hardly matters.